generate/generate.go

```diff
@@ -41,8 +46,12 @@ func BuildStructs(schemaPath string, outputDir string, useMessagePack bool) erro
 //
 // The nameMap allows specifying the type name of the structure for each JSON file.
 //
+// The fieldNameMap is used to provide alternate names for fields in the resulting structs.
+// The property names in the JSON tags for these structs remains the same as supplied.
+// This can be used to accommodate names that are valid JSON but not valid Go identifiers
+//
 // NOTE: If oneOf/allOf entries exist than any JSON schema instances in the root schema file will be skipped.
-func BuildStructsRename(schemaPath string, outputDir string, useMessagePack bool, nameMap map[string]string) error {
+func BuildStructsRenameFields(schemaPath string, outputDir string, useMessagePack bool, nameMap map[string]string, fieldNameMap map[string]string) error {
```

Comment:
This function now has 5 arguments. Have we reached the point where we should have struct for the arguments rather than continuing to make the function signature larger?

Reply:
I'll see what I can do